src/HookRegistry.php

```diff
@@ -33,8 +34,15 @@ public function __construct( Store $store, Options $options ) {
 	 * @since  1.0
 	 */
 	public function register() {
-		foreach ( $this->handlers as $name => $callback ) {
-			Hooks::register( $name, $callback );
+		if ( class_exists( 'MediaWiki\HookContainer\HookContainer' ) ) {
```

Comment:
Using HookContainer::class is better. This works even if the class does not exist.

Reply:
If the check needed at all? I see HookContainer was added in MW 35

Reply:
> If the check needed at all? I see HookContainer was added in MW 35

No it is not needed.